### PR TITLE
 fix splatterplot not in focus issue

### PR DESCRIPTION
**Pull request by AntonPiccardoSelg**
This fixes issue [#11677](http://trac.mantidproject.org/mantid/ticket/11677)
### For Testing

**Context**

When removing the peaksFilter in the splatterplot view, the underlying splatterplot filter was not made visible. It seems that there has been an API change in ParaView and that using pqDeleteReaction instead of the ObjectBuilder destroying the filter is the right thing to do.

Another issue is that when the peaks table was deleted, the color of the peaks did not revert back to white.

The required MDEvent data set can be found [here](http://trac.mantidproject.org/mantid/attachment/ticket/11677/MDEvent_Osiris.nxs) (EWS) and the Peaks data set [here](http://trac.mantidproject.org/mantid/attachment/ticket/11677/MDPeaks_NaCl_1.nxs) (PWS).

**Ensuring that the splatterplot filter is visible**
1. Load EWS into the VSI
2. Switch to the splatterplot mode
3. Drag and drop the PWS into the VSI
4. Click on Peaks->Show all peaks in table
   - Confirm the peaks table is created. 
   - Confirm that the MDPeaksFilter is created
   - Confirm that the color of the peaks has changed (should be redish)
5. Click Peaks-> Remove table
   - Confirm that the table was removed.
   - Confirm that the color of the peaks has switched back to white
   - **Confirm that the splatterplot filter is visible**
6. Apply step 4 again
7. Manually delete the MDPeaksFilter
   - **Confirm that the splatterplot filter is visible**
8. Switch to standard view and back to Splatterplot
   - Confirm that only the original source and the splatterplot filter are loaded

### CODE REVIEW

File: Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/SplatterPlotView.cpp [hunk 1]
```diff
@@ -466,8 +468,7 @@ void SplatterPlotView::onRemovePeaksTable()
 
   if (m_peaksFilter)
   {
-    pqObjectBuilder *builder = pqApplicationCore::instance()->getObjectBuilder();
-    builder->destroy(m_peaksFilter);
+    this->destroyFilter(QString("MDPeaksFilter"));
```
Comment (OwenArnold) [inline Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/SplatterPlotView.cpp]: Lets use static const's for these repeated identifier names, also, shouldn't need to explicitly construct a QString like this.

File: Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/SplatterPlotView.cpp [hunk 2]
```diff
@@ -523,7 +524,7 @@ void SplatterPlotView::createPeaksFilter()
     // Destroy peak filter
     if (m_peaksFilter)
     {
-      builder->destroy(m_peaksFilter);
+      this->destroyFilter(QString("MDPeaksFilter"));
```
Comment (OwenArnold) [inline Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/SplatterPlotView.cpp]: Same as above use static const for identifier

File: Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/SplatterPlotView.cpp [hunk 3]
```diff
@@ -631,8 +632,7 @@ void SplatterPlotView::updatePeaksFilter(pqPipelineSource* filter) {
 
   // If there are no peaks, then destroy the filter, else update it.
   if (peaksSource.isEmpty()) {
-    pqObjectBuilder* builder = pqApplicationCore::instance()->getObjectBuilder();
-    builder->destroy(filter);
+    this->destroyFilter(QString("MDPeaksFilter"));
```
Comment (OwenArnold) [inline Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/SplatterPlotView.cpp]: and here

File: Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/StandardView.cpp [hunk 4]
```diff
@@ -121,7 +121,7 @@ void StandardView::setupViewButtons()
 void StandardView::destroyView()
 {
   pqObjectBuilder *builder = pqApplicationCore::instance()->getObjectBuilder();
-  this->destroyFilter(builder, QString("Slice"));
+  this->destroyFilter(QString("Slice"));
```
Comment (OwenArnold) [inline Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/StandardView.cpp]: and here ...

File: Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/StandardView.cpp [hunk 5]
```diff
@@ -270,8 +270,8 @@ void StandardView::setRebinAndUnbinButtons()
   bool allowRebinning = numberOfTrueMDHistoWorkspaces > 0 || numberOfPeakWorkspaces > 0;
   this->allowRebinningOptions(allowRebinning);
 
-  // If there are no temporary workspaces the button should be disabled.
-  const bool allowUnbin = !( numberOfTemporaryWorkspaces == 0 );
+  // If there are no internally rebinned workspaces the button should be disabled.
+  const bool allowUnbin = !( numberOfInternallyRebinnedWorkspaces == 0 );
   allowUnbinOption(allowUnbin);
```
Comment (OwenArnold) [inline Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/StandardView.cpp]: allowUnbinOption(numberOfInternallyRebinnedWorkspaces > 0);

File: Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/StandardView.cpp [hunk 6]
```diff
@@ -238,12 +238,12 @@ void StandardView::closeSubWindows()
...
  * Note that for a rebin button to be visible there may be no
- * MDHisto workspaces present, yet temporary MDHisto workspaces are
- * allowed.
+ * MDHisto workspaces present, yet  MDHisto workspaces which result from 
+ * rebinning within the VSI are allowed.
  */
 void StandardView::setRebinAndUnbinButtons()
 {
-  int numberOfTemporaryWorkspaces = 0;
+  int numberOfInternallyRebinnedWorkspaces = 0;
```
Comment (OwenArnold) [inline Code/Mantid/Vates/VatesSimpleGui/ViewWidgets/src/StandardView.cpp]: all of these should be unsigned types

**Comment by OwenArnold** [comment]
Also, if you set a background colour in the vsi (with the default background colour setting set in the Mantid preferences), and then drag and drop a peaks workspaces onto the VSI, the colour reverts back to the Mantid preferences version.

**Comment by picatess** [comment]
1. The identifier names of our plugins in splatterplot were moved to MdConstants
2. The suggested for the unsigned ints were implemented
3. The bug regarding the background color was fixed
4. An additional (old) crash bug was discovered and fixed

Parts 3 and 4 require additional testing

__Testing Item 3__

1. Ensure that you have enabled the checkbox "Use defaults for the color map and background color" in the preferences window. 
2. Load the sample data set into the VSI, switch to standard view
3. Change the color map and the backgroudn color; make a note of them
4. Load the same workspace a second time (a distinct source is being created)
  * Confirm that the color map and background color have not changed
5. Switch to the ThreeSlieView
  * Confirm that the color map and background color have not changed
6. Close the VSI, but not Mantid and load the sample data set again into the VSI
  * Confirm that the original color map and background color are loaded.

__Testing Item 4__

The crash occured when switching from standard view to splatterplot view without have an active source. This had not been hedged for in the SplatterPlotView's render method.

1. Load the sample data set into the VSI
2. Click on the builtin node of the pipelineBrowser
  * Confirm that there is no active source, by checking that the properties panel does not contain useful information
3. Switch to the splatterplot view
  * Confirm that Mantid does not crash
  * Confirm that a warning is produced in the log.
4. Drag and drop a peaks workspace into the splatterplot view
  * Confirm that Mantid does not crash
  * Confirm that a warning is produced in the log.

**Pull request merged**